ASM/c/blue_warp.c

```diff
@@ -57,38 +67,35 @@ int32_t DoorWarp1_IsShadowRewardObtained(void) {
 }
 
 void DoorWarp1_KokiriEmerald_Overwrite(void) {
-    z64_file.skybox_time = z64_file.day_time = 0x8000; // CLOCK_TIME(12, 00)
+    // code to run the first time the Queen Gohma blue warp is taken
 }
 
 void DoorWarp1_GoronRuby_Overwrite(void) {
-    z64_file.skybox_time = z64_file.day_time = 0x8000; // CLOCK_TIME(12, 00)
+    // code to run the first time the King Dodongo blue warp is taken
 }
 
 void DoorWarp1_ZoraSapphire_Overwrite(void) {
-    z64_file.skybox_time = z64_file.day_time = 0x8000; // CLOCK_TIME(12, 00)
+    // code to run the first time the Barinade blue warp is taken
 }
 
 void DoorWarp1_ForestMedallion_Overwrite(void) {
-    z64_file.skybox_time = z64_file.day_time = 0x8000; // CLOCK_TIME(12, 00)
+    // code to run the first time the Phantom Ganon blue warp is taken
 }
 
 void DoorWarp1_FireMedallion_Overwrite(void) {
-    z64_file.skybox_time = z64_file.day_time = 0x8000; // CLOCK_TIME(12, 00)
+    // code to run the first time the Volvagia blue warp is taken
     z64_file.event_chk_inf[2] |= 1 << 15; // DMT cloud circle no longer fire
-    z64_file.timer_1_state = 0; // reset heat timer
 }
 
 void DoorWarp1_WaterMedallion_Overwrite(void) {
-    z64_file.skybox_time = z64_file.day_time = 0x4800; // CLOCK_TIME(6, 45)
+    // code to run the first time the Morpha blue warp is taken
     z64_file.event_chk_inf[6] |= 1 << 9; // Lake Hylia water raised
 }
 
 void DoorWarp1_SpiritMedallion_Overwrite(void) {
-    extended_savectx.collected_dungeon_rewards[6] = true;
```

Comment:
~~Your changes mean that this flag and the one for the shadow medallion no longer end up in the extended save context. Intentional?~~

Have found [line 43](https://github.com/OoTRandomizer/OoT-Randomizer/pull/2287/files#diff-bd33fdfebde70839cb2556c496c97e679493d56e67f1ff97ced31bf10e401d8dL43) and am now wondering why setting the flag in this location ever was necessary...

Reply:
I believe this is the cause of the time-of-day bug introduced in #2251.